openedx/features/enterprise_support/api.py

```diff
@@ -311,34 +353,34 @@ def enterprise_customer_for_request(request):
             # Check if there's an Enterprise Customer such that the linked SSO provider
             # has an ID equal to the ID we got from the running pipeline or from the
             # request tpa_hint URL parameter.
-            ec_uuid = EnterpriseCustomer.objects.get(
+            enterprise_customer_uuid = EnterpriseCustomer.objects.get(
                 enterprise_customer_identity_provider__provider_id=sso_provider_id
             ).uuid
         except EnterpriseCustomer.DoesNotExist:
             # If there is not an EnterpriseCustomer linked to this SSO provider, set
             # the UUID variable to be null.
-            ec_uuid = None
+            enterprise_customer_uuid = None
     else:
         # Check if we got an Enterprise UUID passed directly as either a query
         # parameter, or as a value in the Enterprise cookie.
-        ec_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(settings.ENTERPRISE_CUSTOMER_COOKIE_NAME)
-
-    if not ec_uuid and request.user.is_authenticated():
-        # If there's no way to get an Enterprise UUID for the request, check to see
-        # if there's already an Enterprise attached to the requesting user on the backend.
-        learner_data = get_enterprise_learner_data(request.site, request.user)
-        if learner_data:
-            ec_uuid = learner_data[0]['enterprise_customer']['uuid']
-    if ec_uuid:
+        enterprise_customer_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(
+            settings.ENTERPRISE_CUSTOMER_COOKIE_NAME
+        )
+
+    if enterprise_customer_uuid:
```

Comment:
It is possible that you have already retrieved the `EnterpriseCustomer` above. If so, just return the serialized object instead of making an API call to get the `EnterpriseCustomer` again.

Reply:
@douglashall You are right that we could just return the enterprise customer immediately after retrieving it above but this method is also being used by the the LMS from the method `enterprise_sidebar_context` which needs the enterprise branding configuration. Since the `enterprise-customer` endpoints provides us the serialized branding configurations we need to make making an API call to get the EnterpriseCustomer.

The proper solution for this is added as **FIXME** (Implement an Enterprise API endpoint where we can get the EC directly via the linked SSO provider). We would require separate ticket to tackle that.

Reply:
Okay. We can address this function when we tackle https://openedx.atlassian.net/browse/ENT-660.